include/operations/blas_trees.hpp

```diff
@@ -216,18 +173,35 @@ struct StrideExpr {
   size_t N;
 
   StrideExpr(RHS &_r, long offt, long strd, size_t N)
-      : r(_r), offt(offt), strd(strd), N(N) {}
+      : r(_r), offt(offt), strd(strd), N(N)
+  {}
 
-  StrideExpr(BreakIfExpr<RHS, MakePointer> &_br, long offt, long strd, size_t N)
-      : StrideExpr(_br.r, offt, strd, N) {}
+  StrideExpr(RHS &&_r, long offt, long strd, size_t N):
+    StrideExpr(_r, offt, strd, N)
+  {}
 
   size_t getSize() const { return N; }
 };
 
+template <typename RHS> struct strdExpr_constructor {
+  using BRHS = BreakExpr<RHS, MakeHostPointer>;
+  using rettype = StrideExpr<BRHS, MakeHostPointer>;
+  static rettype make(RHS &r, long offset, long stride, size_t N) {
+    return rettype(BRHS(r, !(offset==0&&stride==1&&N==r.getSize())), offset, stride, N);
+  }
+};
+
+template <typename ScalarT, typename ContainerT> struct strdExpr_constructor<vector_view<ScalarT, ContainerT>> {
+  using RHS = vector_view<ScalarT, ContainerT>;
+  using rettype = StrideExpr<RHS, MakeHostPointer>;
+  static rettype make(RHS &r, long offset, long stride, size_t N) {
+    return rettype(r, offset, stride, N);
+  }
+};
+
```

Comment:
The routine fails if different stride or offset is used in related operations on the same vector.

Reply:
Fixed